hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/mapreduce/MapReduceBackupCopyJob.java

```diff
@@ -331,12 +331,12 @@ protected Path createInputFileListing(Job job) throws IOException {
     private Text getKey(Path path) {
       int level = conf.getInt(NUMBER_OF_LEVELS_TO_PRESERVE_KEY, 1);
       int count = 0;
-      String relPath = "";
+      StringBuilder relPath = new StringBuilder();
```

Comment:
It's not that obvious to me that using StringBuilder in this way is faster (looking at the jira, I assumed this is the purpose of the change) than simple string concatenation, because StringBuilder.insert() may involve multiple array copy.

How about instantiating StringBuilder with a capacity (=level) as parameter so that we don't need to re-allocate & copy arrays?

Reply:
But the length of path.getName() is not fixed, how to determine the initial capacity?

Reply:
Maybe a possible way is to store the path name in a list and then concat the list of strings from tail to head?

Reply:
But is it acceptable to apply for the space of the collection and the time cost of traversing the collection again?

Reply:
@Apache9 I changed to use a collection to store strings, PTAL

Reply:
looks good to me. Thank you.